Filtered_kernel/include/CGAL/internal/Static_filters/Do_intersect_3.h

```diff
@@ -128,6 +129,68 @@ class Do_intersect_3
 
 
 
+  result_type
+  operator()(const Tetrahedron_3 &t, const Bbox_3& b) const
+  {
+    Get_approx<Point_3> get_approx;
+    double px, py, pz;
+
+    {
+      const Point_3& p = t[0];
+      if (fit_in_double(get_approx(p).x(), px) && fit_in_double(get_approx(p).y(), py) &&
+          fit_in_double(get_approx(p).z(), pz) )
+        {
+          if( (px >= b.xmin() && px <= b.xmax()) && (py >= b.ymin() && py <= b.ymax())  && (pz >= b.zmin() && pz <= b.zmax()) ){
+            return true;
+          }
+        }else{
+        return Base::operator()(t,b);
```

Comment:
Don't you want to check the next point before giving up? Or do you consider that if one point has ugly coordinates, it is likely that they all do?
(you could use a loop)

Reply:
You are maybe right. And thanks for reading.   Before this commit the first block was slightly different. I also thought maybe a lambda.

Reply:
On the other hand if for one point  the coordinates did not fit into a double  the chances are high that it is the same for the other points.

I was also wondering if it makes sense to check if a corner of the bbox is inside the tetrahedron.